File: LatinReader/app/src/main/java/com/ericmschmidt/latinreader/datamodel/Library.java
package com.ericmschmidt.latinreader.datamodel;

import java.util.ArrayList;

/**
 * Contains information about the texts included in the reader.
 */
public class Library {

    private ArrayList<WorkInfo>  _workInfos;

    /**
     * Creates an instance of the Library class.
     */
    public Library(ArrayList<WorkInfo> collection) {
        this._workInfos = collection;
    }

    /**
     * Gets the WorkInfo for the specified work.
     * @param id the ID of the work to get.
     * @return WorkInfo object.
     */
    public WorkInfo getWorkInfoByID(String id) {
        WorkInfo workToGet = null;
        for (WorkInfo i : _workInfos) {
            if (i.getId().equals(id)) {
                workToGet = i;
            }
        }
        return workToGet;
    }

    /**
     * Gets the list of works as a series of strings.
     * @return String[]
     */
    public String[] getWorkList() {
        String[] works = new String[this._workInfos.size()];

        for (int i = 0; i < this._workInfos.size(); i++) {
            works[i] = this._workInfos.get(i).getTitle();
        }
        return works;
    }

    /**
     * Gets information about all of the texts in the work.
     * @return WorkInfo[]
     */
    public WorkInfo[] getWorks() {
        WorkInfo[] works = new WorkInfo[this._workInfos.size()];
        return this._workInfos.toArray(works);
    }
}

File: LatinReader/core/src/main/java/com/ericmschmidt/classicsreader/datamodel/Library.kt
package com.ericmschmidt.classicsreader.datamodel

/** Contains information about the texts included in the reader.
 * @author Eric Schmidt
 * @author <a href="https://telpirion.com">...</a>
 * @version 2.0
 * @since 1.0
 */
class Library(val collection: ArrayList<WorkInfo>) {

    /**
     * Gets the WorkInfo for the specified work ID.
     */
    fun getWorkInfoByID(id: String?): WorkInfo? {
        var workToGet: WorkInfo? = null
        for (i in collection) {
            if (i.id == id) {
                workToGet = i
            }
        }
        return workToGet
    }

    /**
     * Gets the list of works as a series of strings.
     */
    fun getWorkList(): Array<String?> {
        val works = arrayOfNulls<String>(this.collection.size)

        for (i in this.collection.indices) {
            works[i] = this.collection[i].title
        }
        return works
    }

    /**
     * Gets information about all of the texts in the work.
     */
    fun getWorks(): Array<WorkInfo?> {
        val works = arrayOfNulls<WorkInfo>(this.collection.size)
        return this.collection.toArray<WorkInfo?>(works)
    }
}